assets/components/ModalFilter.js

```diff
@@ -1,28 +1,43 @@
+import CheckBox from "./CheckBox.js";
+
 const ModalFilter = (categories) => {
   return `
-  <div
-    style="display:none"
-    class="modal-filter">
-    <button 
-      class="modal-filter--close">X</button>
-    <div>
-      ${ categories?.map(item => {
-        return `
-          <h2>${ item?.name }</h2>
-          ${ 
-            item?.options?.map(option => {
-              return `
-                <input
-                  type="checkbox"
-                  data-filter-for="${ option?.name?.toLowerCase() }"
-                  class="filter-cb-${ option?.name?.toLowerCase() }" />
-                ${ option?.name }
-              `
+    <div
+      class="modal-filter" style="display:none;">
+      <div class="modal-filter--top">
+        <button 
+          class="modal-filter--close-btn">
+          <img src="/assets/icons/close.svg"/>
+        </button>
+        <div class="modal-filter--heading-box">
+          <h2 class="modal-filter--heading-txt">Filters</h2>
+        </div>
+      </div>
+      <div class="modal-filter--wrapper-outer">
+        <div class="modal-filter--wrapper-inner">
+          ${ categories?.map(item => {
+            return `
+              <div class="modal-filter--catergories-wrapper-outer">
+                <h3>${ item?.name }</h3>
+                <div class="modal-filter--categories-wrapper-inner">
+                  ${ 
+                    item?.options?.map(option => {
+                      return  CheckBox(option?.name?.toLowerCase(), option?.name);
```

Comment:
I think you might want to send the option id to the checkbox in the first argument here.